crates/starknet-types-core/src/felt/mod.rs

```diff
@@ -230,6 +273,17 @@ impl Felt {
         alloc::format!("{self:#x}")
     }
 
+    /// Helper to produce a hexadecimal formatted string of 66 chars.
+    #[cfg(feature = "alloc")]
+    pub fn to_fixed_hex_string(&self) -> alloc::string::String {
+        let hex_str = alloc::format!("{self:#x}");
+        if hex_str.len() < 66 {
+            alloc::format!("0x{:0>64}", hex_str.strip_prefix("0x").unwrap())
+        } else {
+            hex_str
+        }
```

Comment:
Are you 100% sure this can't been expressed in one single `format` call? It's powerfull

Reply:
it doesn't work and i think it's because of the way the `ToHex` is implemented